### DISCUSSION TITLE
Erro no html da aula sobre validators

### DISCUSSION THREAD

**Discussion by brunoolympioferreira** [balta-io/7187#69] [Perguntas e Respostas] [answered]
Meu código está identico ao curso, porém o compilador de html diz que a varíavel username é inválida, estou usando o angular 13

![erroCursoBalta](https://user-images.githubusercontent.com/79343105/151816353-42e77da1-d45f-499e-926f-74cf2caa90be.PNG)

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @brunoolympioferreira, poderia colocar o código do seu TypeScript por favor!

**Reply by brunoolympioferreira** [reply]
`import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { DataService } from 'src/app/services/data.service';

@Component({
  selector: 'app-login-page',
  templateUrl: './login-page.component.html',
  styleUrls: ['./login-page.component.css']
})
export class LoginPageComponent implements OnInit {
  public form: FormGroup;

  constructor(
    private service: DataService,
    private fb: FormBuilder
  ) {
    this.form = this.fb.group({
      username: ['', Validators.compose([
        Validators.minLength(14),
        Validators.maxLength(14),
        Validators.required,
      ])],
      password: ['', Validators.compose([
        Validators.minLength(6),
        Validators.maxLength(20),
        Validators.required
      ])]
    });
  }

  ngOnInit(): void {
  }

}
`

**Reply by andrebaltieri** [reply]
@brunoolympioferreira está dando erro quando você executa a aplicação ou só aparece essa observação no VSCode?
Ao que me parece seu arquivo .TS está correto!

**Comment by hosanah** [comment]
Tô com esse mesmo erro, consigo achar essas variáveis no developer tools do Chrome, porém na aplicação me tras esse erro

error TS4111: Property 'usuario' comes from an index signature, so it must be accessed with ['usuario']

**Reply by hosanah** [reply]
Não sei se está correto, mas corrigi esse erro setando essa propriedade no tsconfig.json pra false.

"noPropertyAccessFromIndexSignature": false,

Porém, minha aplicação agora está quebrada me trazer o seguinte erro: NG01050

**Reply by hosanah** [reply]
Esse erro NG01050, foi devido ao bind do FormGroup que não estava assim [formGroup]

**Reply by andrebaltieri** [reply]
Bom dia, @hosanah, poderia levar esta questão para nosso fórum?
https://github.com/balta-io/forum/discussions
